Scripts/Runtime/Util/UnsafeUtil.cs

```diff
@@ -0,0 +1,30 @@
+using Unity.Collections.LowLevel.Unsafe;
+
+namespace Anvil.Unity.DOTS.Util
+{
+    /// <summary>
+    /// A collection of utilities that help with unsafe work.
+    /// </summary>
+    public unsafe class UnsafeUtil
+    {
+        /// <summary>
+        /// Determines whether two <see cref="{T}"/> instances are equal without boxing.
+        /// </summary>
+        /// <param name="value1">The first <see cref="{T}"/> compare.</param>
+        /// <param name="value2">The second <see cref="{T}"/> compare.</param>
+        /// <returns>True if the <see cref="{T}"/>s are the same.</returns>
+        /// <remarks>
+        /// Useful for situations where <see cref="System.IEquatable{T}"/> can't be implemented
+        /// (Ex: 3rd-party assemblies)
+        /// </remarks>
+        public static bool Equals_NoBox<T>(in T value1, in T value2)
+            where T : unmanaged
+        {
+            return UnsafeUtility.MemCmp(
+                    UnsafeUtilityExtensions.AddressOf(in value1),
+                    UnsafeUtilityExtensions.AddressOf(in value2),
+                    UnsafeUtility.SizeOf<T>())
+                == 0;
+        }
```

Comment:
I get wanting to use `in` to signify that this method won't change anything but odds are pretty good that the type of `T` won't always be a `readonly` struct. If it's not a `readonly` struct then there will be a defensive copy. 

`ref` is less clear and allows us to modify if we want to, but it does guarantee we're operating on the original address and there isn't a copy going on. Almost all of Unity's API's are using `ref` instead of `in` for the same reason.

```suggestion
        public static bool Equals_NoBox<T>(ref T value1, ref T value2)
            where T : unmanaged
        {
            return UnsafeUtility.MemCmp(
                    UnsafeUtility.AddressOf(ref value1),
                    UnsafeUtility.AddressOf(ref value2),
                    UnsafeUtility.SizeOf<T>())
                == 0;
        }
```

Reply:
See other rationale. There won't be a defensive copy.